tasks/Task.cpp

```diff
@@ -28,27 +28,265 @@ bool Task::configureHook()
+
+    // Set the pose transformation between the camera and the robot frame
+    Eigen::Affine3d camera2body;
+    _camera2body.get(base::Time::now(), camera2body);
+
+    //Computes the Velocity Twist Matrix from the spatial transformation between the camera and the robot
+    vpTranslationVector cto(camera2body.translation()[0],
+                            camera2body.translation()[1],
+                            camera2body.translation()[2]);
+    vpRxyzVector cro(camera2body.rotation().eulerAngles(0,1,2)[0],
+                     camera2body.rotation().eulerAngles(0,1,2)[1], 
+                     camera2body.rotation().eulerAngles(0,1,2)[2]);
+    vpRotationMatrix cRo(cro); // Build the rotation matrix
+
+    //build the homogeneous matrix
+    cMe.buildFrom(cto, cRo); 
+    //build and set the velocity twist matrix
+    cVe.buildFrom(cMe);
+    task.set_cVe(cVe);
+
     return true;
 }
+
 bool Task::startHook()
 {
     if (! TaskBase::startHook())
         return false;
     return true;
 }
+
 void Task::updateHook()
 {
     TaskBase::updateHook();
+
+    std::vector<base::Vector2d> corners;
+    _marker_corners.read(corners);
+
+    if(state() == RUNNING)
+    {
+        state(WAITING_FIRST_MEASUREMENTS);
+    }
+
+    if (corners.size() > 0 && state() == WAITING_FIRST_MEASUREMENTS)
+    {
+        // Sets the desired position of the visual feature (reference point)
+        if (updateDesiredPose())
+            return;
+        updateFeatures(corners);
+        state(CONTROLLING);
+    }
+    else if (corners.size() > 0 && state() == CONTROLLING)
+    {
+        if (updateDesiredPose())
+            return;
+        updateFeatures(corners);
+
+        // Update this jacobian in the task structure. It will be used to compute
+        // the velocity skew (as an articular velocity)
+        // qdot = -lambda * L^+ * cVe * eJe * (s-s*)
+        task.set_eJe(eJe) ;
+
+        // Set the gain
+        setGain();
+
+        // Compute the visual servoing skew vector
+        vpColVector v;
+        task.set_cVe(cVe);
+        v = task.computeControlLaw() ;
+
+        writeVelocities(v);
+        corners.clear();
+    }
 }
+
 void Task::errorHook()
 {
     TaskBase::errorHook();
 }
+
 void Task::stopHook()
 {
     TaskBase::stopHook();
 }
+
 void Task::cleanupHook()
 {
     TaskBase::cleanupHook();
+    
+    task.kill();
+}
+
+void Task::updateFeatures(std::vector<base::Vector2d> corners)
+{
+    //  Compute the initial pose
+    pose.clearPoint();
+
+    for (int i=0 ; i < 4 ; i++)
+    {
+        base::Vector2d aux;
+        aux = corners[i];
+        //vpImagePoint is instanciated using (v,u) instead of (u,v)
+        vpImagePoint a(aux[1],aux[0]);
+        double x,y;
+        vpPixelMeterConversion::convertPoint(cam_cal,a,x,y);
+        point[i].set_x(x);
+        point[i].set_y(y);
+        pose.addPoint(point[i]) ;
+    }
+
+    pose.computePose(vpPose::LAGRANGE, cMo);
+    pose.computePose(vpPose::VIRTUAL_VS, cMo);
```

Comment:
No...the first one computes the pose and the second one is a refinement that improves the accuracy.